File: src/main/java/org/example/attendance/entity/Employee.java
package org.example.attendance.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "employee")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class Employee {

    public enum Role { EMPLOYEE, ADMIN }
    public enum Status { ACTIVE, INACTIVE }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "employee_no", nullable = false, unique = true, length = 32)
    private String employeeNo;

    @Column(nullable = false, length = 100)
    private String name;

    @Column(length = 20)
    private String phone;

    @Column(unique = true, length = 200)
    private String email;

    @Column(name = "password_hash", nullable = false, length = 255)
    private String passwordHash;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Role role = Role.EMPLOYEE;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Status status = Status.ACTIVE;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "department_id")
    private Department department;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "office_location_id")
    private OfficeLocation officeLocation;

    @Column(name = "avatar_url", length = 500)
    private String avatarUrl;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}

File: attendance_app/lib/models/employee.dart
class Employee {
  final int id;
  final String employeeNo;
  final String name;
  final String? phone;
  final String? email;
  final String role;
  final String status;
  final String? departmentName;
  final int? departmentId;
  final String? officeLocationName;
  final int? officeLocationId;
  final String? avatarUrl;

  const Employee({
    required this.id,
    required this.employeeNo,
    required this.name,
    this.phone,
    this.email,
    required this.role,
    required this.status,
    this.departmentName,
    this.departmentId,
    this.officeLocationName,
    this.officeLocationId,
    this.avatarUrl,
  });

  bool get isAdmin => role == 'ADMIN';
  bool get isActive => status == 'ACTIVE';

  factory Employee.fromJson(Map<String, dynamic> json) => Employee(
        id: (json['id'] as num).toInt(),
        employeeNo: json['employeeNo'] as String,
        name: json['name'] as String,
        phone: json['phone'] as String?,
        email: json['email'] as String?,
        role: json['role'] as String,
        status: json['status'] as String,
        departmentName: json['departmentName'] as String?,
        departmentId: (json['departmentId'] as num?)?.toInt(),
        officeLocationName: json['officeLocationName'] as String?,
        officeLocationId: (json['officeLocationId'] as num?)?.toInt(),
        avatarUrl: json['avatarUrl'] as String?,
      );
}
